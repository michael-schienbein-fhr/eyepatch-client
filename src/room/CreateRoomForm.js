import { useState, useContext, useEffect } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../common/Alert";
import UserContext from "../auth/UserContext";
import EyepatchApi from "../api/api";
import jwt from "jsonwebtoken";

function CreateRoomForm({ createRoom }) {
  const { currentUser } = useContext(UserContext);
  // const [passFlag, setPassFlag] = useState(null);
  const history = useHistory();
  const [formData, setFormData] = useState({
    room_owner: "",
    room_name: "",
    password: "",
  });
  const [newRoomId, setNewRoomId] = useState(null);
  const [formErrors, setFormErrors] = useState([]);
  useEffect(function updateRoomInfoOnMount() {

    async function getRoomId() {
      try {
        let newRoom = await EyepatchApi.getNewest();
        setNewRoomId(newRoom.id);
      } catch (err) {
        console.error("App getRoomId: problem loading", err);
      }

    }
    getRoomId();
    setFormData(d => ({ ...d, room_owner: currentUser.username }));
  }, []);

  async function handleSubmit(evt) {
    evt.preventDefault();
    try {
      let result = await createRoom(formData);
      let { passFlag } = jwt.decode(result.roomToken);
      console.log(result, passFlag);
      if (result.success && passFlag === false) {
        history.push(`/rooms/${newRoomId + 1}`);
      } else if (result.success && passFlag === true) {
        history.push(`/rooms/private/${newRoomId + 1}`);
      } else {
        setFormErrors(result.errors);
      }
    } catch (err) {
      console.error(err);
    }



  }

  /** Update form data field */
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
    <div className="SignupForm">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h2 className="mb-3">Create Room</h2>
        <div className="card">
          <div className="card-body">
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <label>Room Name</label>
                <input
                  name="room_name"
                  className="form-control"
                  value={formData.room_name}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <label>Password</label>
                <input
                  type="password"
                  name="password"
                  className="form-control"
                  value={formData.password}
                  onChange={handleChange}
                />
              </div>
              {formErrors.length
                ? <Alert type="danger" messages={formErrors} />
                : null
              }

              <button
                type="submit"
                className="btn btn-md btn-outline-secondary mt-3"
                onSubmit={handleSubmit}
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default CreateRoomForm;