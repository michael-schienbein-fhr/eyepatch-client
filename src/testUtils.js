import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const demoQueueVideo = {
  username: 'testuser2',
  type: 'video',
  action: 'add',
  text: '"testVideoTitle" added to queue for user: "testuser2".',
  videoId: 'testVideoId',
  title: 'testVideoTitle',
  description: 'testDescription',
  thumbnail: 'testThumbnail'
}
const demoSearchVideo = { "kind": "youtube#searchResult", "etag": "3_5Vn-wWRzIN5QC7aOEWvZsmMwI", "id": { "kind": "youtube#video", "videoId": "GjbibR6DV1s" }, "snippet": { "publishedAt": "2015-08-11T17:03:57Z", "channelId": "UCi_ZH3bYHidyHsETH5hbTFQ", "title": "Dusky - Skin Deep (Official Video)", "description": "Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \"This EP was all ...", "thumbnails": { "default": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/default.jpg", "width": 120, "height": 90 }, "medium": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg", "width": 320, "height": 180 }, "high": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/hqdefault.jpg", "width": 480, "height": 360 } }, "channelTitle": "DuskyMusic", "liveBroadcastContent": "none", "publishTime": "2015-08-11T17:03:57Z" } }

const testRoom = {
  "id": 1,
  "roomOwner": "testadmin",
  "roomName": "testadmin",
  "roomMembers": null,
  "videoQueue": null,
  "hasPass": true,
  "created_at": "2021-10-22T10:34:59.443Z"
}
const UserProvider =
  ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider, demoUser, demoQueueVideo, demoSearchVideo, testRoom };

