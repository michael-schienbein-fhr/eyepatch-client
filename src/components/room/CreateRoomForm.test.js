import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { UserProvider, demoQueueVideo, demoSearchVideo } from "../../testUtils";
import CreateRoomForm from "./CreateRoomForm";

// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";
// const nock = require('nock')



it("matches snapshot", function () {

  const { asFragment } = render(
    <UserProvider>
      <CreateRoomForm/>
    </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
