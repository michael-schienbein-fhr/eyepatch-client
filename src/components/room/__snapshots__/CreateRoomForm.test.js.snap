// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        Create Room
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label>
                Room Name
              </label>
              <input
                class="form-control"
                name="room_name"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Password
              </label>
              <input
                class="form-control"
                name="password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-md btn-outline-secondary mt-3"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
